src/stactools/sentinel1/grd/product_metadata.py

```diff
@@ -42,7 +50,7 @@ def __init__(
 
         self.resolution = self.product_id.split("_")[2][-1]
 
-        def _get_geometries():
+        def _get_geometries() -> Tuple[List[float], Dict[str, Any]]:
             # Find the footprint descriptor
             footprint_text = self._root.findall(".//gml:coordinates")
```

Comment:
`findall` will _never_ return `None` (it returns an _empty list_ if nothing is found), so the subsequent `None` check is useless. However, since what you're really looking for is the `text` value of the _first_ found element, the `None` check becomes useful when you change the line above to this:

```suggestion
            footprint_text = self._root.find_text(".//gml:coordinates")
```

This also makes the variable name more appropriate for the value it represents.